src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -296,14 +336,66 @@ public void SetOptions(DacDeployOptions options)
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-
+                // Note that we set excluded object types here since dacfx has this value as null;
                 if (prop != null)
                 {
-                    deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
                 }
             }
         }
 
+        /// <summary>
+        /// Sets the Value and Description to all properties
+        /// </summary>
+        /// <param name="prop"></param>
+        /// <param name="val"></param>
+        /// <param name="deployOptionsProp"></param>
+        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        {
+            var val = prop.GetValue(options);
+            var attribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault();
+
+            // DeploymentOptionProperty is a generic type, which requires value type to intitialize 
```

Comment:
Couldn't you just use Object as the type instead since you don't really care about the specific type when setting it?

Reply:
Yes, we really don't care what type of value the property has, we mostly focusing on bool and other types which are not null! In such case I am going to revert the generic type to just an object, since setting object type to boolean or something would be an exception case unless typecasting. or is there any other approach to leave as generic and achieve the same with object type!

Reply:
You should leave the class generic since that's useful to have when accessing the properties on the object. I'm just saying here when you're creating it you just create it with the type of object so that it can take anything.

Reply:
Looks like without typecasting work is not going to be done for at least ObjectTypes[]! Since the DeploymentOptionProperty<object> has to set to the Dac Options of <ObjectType[]>.

DeploymentOptionProperty<object> is going to set to DeploymentOptionProperty<bool/string/ObjectType[]> not the reverse way! so I am going to do all the properties as object type except the ObjectType[] which needs a typecasting by verifying with the property type.